src/cluster.h

```diff
@@ -173,257 +53,83 @@ typedef struct {
     redisDb *db;                /* A link back to the db this dict belongs to */
 } clusterDictMetadata;
 
+/*-----------------------------------------------------------------------------
+ * Cluster public types
+ *----------------------------------------------------------------------------*/
+typedef intptr_t clusterNodeHandle;
+
 typedef struct clusterState {
-    clusterNode *myself;  /* This node */
-    uint64_t currentEpoch;
-    int state;            /* CLUSTER_OK, CLUSTER_FAIL, ... */
-    int size;             /* Num of master nodes with at least one slot */
-    dict *nodes;          /* Hash table of name -> clusterNode structures */
-    dict *shards;         /* Hash table of shard_id -> list (of nodes) structures */
-    dict *nodes_black_list; /* Nodes we don't re-add for a few seconds. */
-    clusterNode *migrating_slots_to[CLUSTER_SLOTS];
-    clusterNode *importing_slots_from[CLUSTER_SLOTS];
-    clusterNode *slots[CLUSTER_SLOTS];
-    rax *slots_to_channels;
-    /* The following fields are used to take the slave state on elections. */
-    mstime_t failover_auth_time; /* Time of previous or next election. */
-    int failover_auth_count;    /* Number of votes received so far. */
-    int failover_auth_sent;     /* True if we already asked for votes. */
-    int failover_auth_rank;     /* This slave rank for current auth request. */
-    uint64_t failover_auth_epoch; /* Epoch of the current election. */
-    int cant_failover_reason;   /* Why a slave is currently not able to
-                                   failover. See the CANT_FAILOVER_* macros. */
+    clusterNodeHandle myself;  /* This node */
```

Comment:
If there is an API, we shouldn't have a reference for myself anywhere, since you should always get it from the API. I also thought the state would become completely opaque as well based on previous conversations.

Reply:
Sure. Will do.